adapter/service/virtualmeter/manager.go

```diff
@@ -433,14 +416,14 @@ func (m *manager) normalizeOutLvlSwitchLevel(level int64, serviceAddr string) (f
 // createVirtualServicesForThing creates a virtual meter and numeric meter services' specifications depending on
 // presence of other services. Currently virtual metering is support for following services:
 // - outlvlswitch.Service.
-func (m *manager) createVirtualServicesForThing(t adapter.Thing) (outLvlSwitchService *fimptype.Service, numericMeterService *fimptype.Service) {
+func (m *manager) createVirtualServicesForThing(t adapter.Thing, group string) (outLvlSwitchService *fimptype.Service, numericMeterService *fimptype.Service) {
 	for _, s := range t.Services("") {
 		switch s.(type) {
 		case outlvlswitch.Service:
 			return Specification(
 					m.ad.Name(),
 					m.ad.Address(),
-					t.Address(),
+					t.Address()+group,
```

Comment:
Do not construct this service_address on your own, copy it from the outlvlswitch.Service based on which you make these.

Reply:
possibly just s.Specification().Address() would be enough.